From bc29f9db1b6b0291fada416d96ce892299f33117 Mon Sep 17 00:00:00 2001
From: solk2 <solk@solk.org.ua>
Date: Wed, 1 Oct 2014 14:01:13 +0300
Subject: [PATCH 03/19] klte-common: libril: Fix RIL_Call structure

* Samsung added call_id as the second byte of index
* Split index to avoid a workaround in RIL subclass
* Add Samsung call detail fields, reorganize to fix
  incorrect structure, to avoid hacks in RIL class

Change-Id: I023228c05165c0cef9b698846a96484a6d318092
---
 include/telephony/ril.h | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/include/telephony/ril.h b/include/telephony/ril.h
index 4b4f715..bb04a2b 100644
--- a/include/telephony/ril.h
+++ b/include/telephony/ril.h
@@ -420,7 +420,10 @@ typedef struct {
 
 typedef struct {
     RIL_CallState   state;
-    int             index;      /* Connection Index for use with, eg, AT+CHLD */
+    char            index;      /* Connection Index for use with, eg, AT+CHLD */
+    char            call_id;    /* Samsung call_id */
+    char            foo1;       /* Samsung */
+    char            foo2;       /* Samsung */
     int             toa;        /* type of address, eg 145 = intl */
     char            isMpty;     /* nonzero if is mpty call */
     char            isMT;       /* nonzero if call is mobile terminated */
@@ -430,8 +433,11 @@ typedef struct {
     char            isVoicePrivacy;     /* nonzero if CDMA voice privacy mode is active */
     char *          number;     /* Remote party number */
     int             numberPresentation; /* 0=Allowed, 1=Restricted, 2=Not Specified/Unknown 3=Payphone */
-    char *          name;       /* Remote party name */
+    char            call_type;  /* Samsung*/
+    char            call_domain;/* Samsung*/
     int             namePresentation; /* 0=Allowed, 1=Restricted, 2=Not Specified/Unknown 3=Payphone */
+    char *          csv;        /* Samsung */
+    char *          name;       /* Remote party name */
     RIL_UUS_Info *  uusInfo;    /* NULL or Pointer to User-User Signaling Information */
 } RIL_Call;
 
-- 
2.7.4

